### PR TITLE
[NU-1740] processing types reload

### PR DESCRIPTION
**Pull request by mk-software-pl**
## Describe your changes

- The whole model can be reloaded with `POST /api/app/processingtype/reload` now - you can use this endpoint to reconfigure Nu processing types without need to restart the app
- refactoring

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Comment by wrzontek** [comment]
>The whole model can be reloaded with POST /api/app/processingtype/reload now - you can use this endpoint to reconfigure Nu processing types without need to restart the app

This essentially boils down to the change that now the application conf is also reloaded from file during `reload`, right?

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/config/UsageStatisticsReportsConfig.scala [hunk 1]
```diff
@@ -2,7 +2,7 @@ package pl.touk.nussknacker.ui.config
 
 final case class UsageStatisticsReportsConfig(
     enabled: Boolean,
-    errorReportsEnabled: Boolean,
+    errorReportsEnabled: Boolean = true,
```
Comment (lukasz-bigorajski) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/config/UsageStatisticsReportsConfig.scala:5]: Why this is changed?

**Review by lukasz-bigorajski — APPROVED** [review]
I'm realy impressed!
LGTM!

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/config/UsageStatisticsReportsConfig.scala (hunk 1 above)
Comment (mk-software-pl) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/config/UsageStatisticsReportsConfig.scala]: Good catch. It should be changed in common-designer.conf instead

**Review by arkadius — APPROVED** [review]
LGTM, one question added

File: designer/server/src/test/resources/config/common-designer.conf [hunk 2]
```diff
@@ -60,6 +60,7 @@ notifications {
 
 usageStatisticsReports {
   enabled: true
+  errorReportsEnabled: true
```
[the remark this answers is not in the document]
Comment (mk-software-pl) [reply designer/server/src/test/resources/config/common-designer.conf:63]: yes, there is no default value for errorReportsEnabled so it should also be configured here. The same is done in `defaultDesignerConfig.conf`

**Pull request merged**